src/yaml/yaml.cr

```diff
@@ -14,7 +15,34 @@ module YAML
   alias Type = String | Hash(Type, Type) | Array(Type) | Nil
   alias EventKind = LibYAML::EventType
 
-  def self.load(data)
+# Serializes a document in YAML format.
+#
+# ```yaml
+# # ./foo.yml
+# foo:
+#   bar: "foobar"
+# ```
+#
+# Returns `Hash` if *data* is in YAML format.
+# ```crystal 
+# require "yaml"
+#
+# YAML.load(File.read("./foo.yml"))
+# #=> {"foo" => {"bar" => "foobar"}}
+# ```
+#
+# Returns the same value if *data* is not in YAML format.
+# ```crystal 
+# YAML.load("foo")
+# #=> "foo"
+# ```
```

Comment:
This simply is a valid YAML string which deserialization just happens to look the same. Try

```yml
1
```
```yml
"foo bar"
```
```yml
- 1
- "foo"
```

and so on.